core/src/bank_forks.rs

```diff
@@ -587,57 +567,103 @@ mod tests {
+            PathBuf::from(snapshot_output_path.path()),
+            100,
+        );
+        bank_forks.set_snapshot_config(snapshot_config.clone());
+        let bank0 = bank_forks.get(0).unwrap();
+        snapshot_utils::add_snapshot(&snapshot_config.snapshot_path, bank0, &vec![]).unwrap();
+        for slot in 0..last_slot {
+            let mut bank = Bank::new_from_parent(&bank_forks[slot], &Pubkey::default(), slot + 1);
+            f(&mut bank, &mint_keypair);
+            let bank = bank_forks.insert(bank);
+            // set root to make sure we don't end up with too many account storage entries
+            // and to allow purging logic on status_cache to kick in
+            bank_forks.set_root(bank.slot(), &None);
+            snapshot_utils::add_snapshot(
                 &snapshot_config.snapshot_path,
-                snapshot_utils::get_snapshot_tar_path(
-                    &snapshot_config.snapshot_package_output_path,
-                ),
+                &bank,
+                &(vec![bank.slot()]),
             )
             .unwrap();
-            SnapshotPackagerService::package_snapshots(&snapshot_package).unwrap();
+        }
+        // Generate a snapshot package for last bank
+        let last_bank = bank_forks.get(last_slot).unwrap();
+        let slot_snapshot_paths =
+            snapshot_utils::get_snapshot_paths(&snapshot_config.snapshot_path);
+        let snapshot_package = snapshot_utils::package_snapshot(
+            last_bank,
+            &slot_snapshot_paths,
+            snapshot_utils::get_snapshot_tar_path(&snapshot_config.snapshot_package_output_path),
+        )
+        .unwrap();
+        SnapshotPackagerService::package_snapshots(&snapshot_package).unwrap();
+
+        restore_from_snapshot(
+            bank_forks,
+            accounts_dir.path().to_str().unwrap().to_string(),
+            last_slot,
+        );
+    }
 
-            restore_from_snapshot(
-                bank_forks,
-                accounts_dir.path().to_str().unwrap().to_string(),
-                index,
+    #[test]
+    fn test_bank_forks_snapshot_n() {
+        run_bank_forks_snapshot_n(4, |bank, mint_keypair| {
+            let key1 = Keypair::new().pubkey();
+            let tx = system_transaction::create_user_account(
+                &mint_keypair,
+                &key1,
+                1,
+                bank.last_blockhash(),
             );
+            assert_eq!(bank.process_transaction(&tx), Ok(()));
+            bank.freeze();
+        });
+    }
+
+    fn goto_end_of_slot(bank: &mut Bank) {
+        let mut tick_hash = bank.last_blockhash();
+        loop {
+            tick_hash = hashv(&[&tick_hash.as_ref(), &[42]]);
+            bank.register_tick(&tick_hash);
+            if tick_hash == bank.last_blockhash() {
+                bank.freeze();
+                return;
+            }
         }
     }
 
+    #[test]
+    fn test_bank_forks_status_cache_snapshot_n() {
```

Comment:
Should we rename this to reflect what the function passed to `run_bank_forks_snapshot_n` does?

Reply:
Or a comment will also do :D